ios/brave-ios/Sources/BraveWallet/Crypto/Stores/PortfolioStore.swift

```diff
@@ -491,6 +495,28 @@ public class PortfolioStore: ObservableObject, WalletObserverStore {
         }
       }
 
+      guard !Task.isCancelled else { return }
+      if selectedAccounts.contains(where: { $0.coin == .btc }) {
+        /// We  need to know if user has pending balance to show/hide banner. Re-fetch on view load.
+        self.accountsBTCBalances = await withTaskGroup(of: [String: [BTCBalanceType: Double]].self)
+        {
+          [bitcoinWalletService] group in
+          for account in selectedAccounts where account.coin == .btc {
+            group.addTask {
+              let btcBalances = await bitcoinWalletService.fetchBTCBalances(
+                accountId: account.accountId
+              )
+              return [account.id: btcBalances]
+            }
+          }
+          var accountsBTCBalances: [String: [BTCBalanceType: Double]] = [:]
+          for await accountBTCBalances in group {
+            accountsBTCBalances.merge(with: accountBTCBalances)
+          }
+          return accountsBTCBalances
+        }
+      }
+
```

Comment:
so right now we are basically calling btc account balance api twice in portfolio since we have one to cache total balance in CD and one for checking if we have pending balance? 
should we just take out the btc total balance caching? since we always need to fetch btc balance in view onAppear?

Reply:
It's called once in cache manager when wallet loads for the global cache, it's not Portfolio specific cache. The cached value is also currently used to determine if user has a BTC account for P3A in `CryptoStore.recordP3AActiveWallets()`. Discussed offline that some places in wallet only display total balance (our cached value), so I think best to keep the caching setup and re-evaluate later if needed.